tests/hypervolume_tests/test_box_decomposition.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+from optuna._hypervolume.box_decomposition import get_non_dominated_box_bounds
+from optuna._hypervolume.wfg import compute_hypervolume
+from optuna.study._multi_objective import _is_pareto_front
+
+
+def _generate_pareto_sols(n_objectives: int, n_trials: int, seed: int) -> np.ndarray:
+    rng = np.random.RandomState(seed)
+    sorted_loss_vals = np.unique(rng.random((n_trials, n_objectives)), axis=0)
+    on_front = _is_pareto_front(sorted_loss_vals, assume_unique_lexsorted=True)
+    return sorted_loss_vals[on_front]
+
+
+@pytest.mark.parametrize("n_objectives", range(2, 5))
+def test_exact_box_decomposition(n_objectives: int) -> None:
```

Comment:
Does this test finish quickly? If not, it would be great to add the `slow` marker to avoid decreasing the test speed.

Reply:
In my environment, I believe the test is quick enough.

```shell
(venv) shuhei@EB-B9400CBA:~/pfn-work/optuna-dev/optuna$ python -m pytest tests/hypervolume_tests/test_box_decomposition.py 
============================= test session starts ==============================
platform linux -- Python 3.11.11, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/shuhei/pfn-work/optuna-dev/optuna
configfile: pyproject.toml
collected 3 items                                                              

tests/hypervolume_tests/test_box_decomposition.py ...                    [100%]

============================== 3 passed in 0.16s ===============================

```